packages/block-library/src/social-links/editor.scss

```diff
@@ -101,19 +101,10 @@
 .wp-block-social-links .block-list-appender {
 	position: static; // display inline.
 
-	.block-editor-button-block-appender.components-button.components-button {
-		padding: $grid-unit-10 - 2px;
-	}
-}
-
-.wp-block-social-links {
-	&.has-small-icon-size .block-editor-button-block-appender.components-button.components-button {
+	.block-editor-button-block-appender {
```

Comment:
From what I've tested, it doesn't seem like we need to increase the CSS specificity too much.